src/predict.rs

```diff
@@ -581,158 +519,52 @@ pub mod test {
     (above, left, o1, o2)
   }
 
-  fn pred_dc_4x4(
-    output: &mut [u16], stride: usize, above: &[u16], left: &[u16]
-  ) {
-    unsafe {
-      highbd_dc_predictor(
-        output.as_mut_ptr(),
-        stride as libc::ptrdiff_t,
-        4,
-        4,
-        above.as_ptr(),
-        left.as_ptr(),
-        8
-      );
-    }
-  }
-
-  fn pred_dc_left_4x4(
-    output: &mut [u16], stride: usize, above: &[u16], left: &[u16]
-  ) {
-    unsafe {
-      highbd_dc_left_predictor(
-        output.as_mut_ptr(),
-        stride as libc::ptrdiff_t,
-        4,
-        4,
-        above.as_ptr(),
-        left.as_ptr(),
-        8
-      );
-    }
-  }
-
-  fn pred_dc_top_4x4(
-    output: &mut [u16], stride: usize, above: &[u16], left: &[u16]
-  ) {
-    unsafe {
-      highbd_dc_top_predictor(
-        output.as_mut_ptr(),
-        stride as libc::ptrdiff_t,
-        4,
-        4,
-        above.as_ptr(),
-        left.as_ptr(),
-        8
-      );
-    }
-  }
-
-  pub fn pred_h_4x4(
-    output: &mut [u16], stride: usize, above: &[u16], left: &[u16]
-  ) {
-    let mut left = left.to_vec();
-    left.reverse();
-    unsafe {
-      highbd_h_predictor(
-        output.as_mut_ptr(),
-        stride as libc::ptrdiff_t,
-        4,
-        4,
-        above.as_ptr(),
-        left.as_ptr(),
-        8
-      );
-    }
-  }
-
-  pub fn pred_v_4x4(
-    output: &mut [u16], stride: usize, above: &[u16], left: &[u16]
-  ) {
-    unsafe {
-      highbd_v_predictor(
-        output.as_mut_ptr(),
-        stride as libc::ptrdiff_t,
-        4,
-        4,
-        above.as_ptr(),
-        left.as_ptr(),
-        8
-      );
-    }
-  }
-
-  pub fn pred_paeth_4x4(
-    output: &mut [u16], stride: usize, above: &[u16], left: &[u16]
-  ) {
-    let mut left = left.to_vec();
-    left.reverse();
-    unsafe {
-      highbd_paeth_predictor(
-        output.as_mut_ptr(),
-        stride as libc::ptrdiff_t,
-        4,
-        4,
-        above.as_ptr(),
-        left.as_ptr(),
-        8
-      );
-    }
-  }
+  macro_rules! wrap_aom_pred_fn {
+    ($fn_4x4:ident, $aom_fn:ident) => {
```

Comment:
Hi- Can you instruct me what 'ident's do here?

Reply:
they are `identfiers`. macros are made so you cannot pass the wrong syntactical element.

[here](https://doc.rust-lang.org/reference/macros-by-example.html) the whole documentation :)